docs/examples/shading/plot_martinez_shade_loss.py

```diff
@@ -0,0 +1,266 @@
+# - The rows have the same true-tracking tilt angles. Let's consider
+#   true-tracking so shade is significant for this example.
+# - Terrain slope is 7 degrees downward to the east.
+# - Rows' axes are horizontal.
+# - The modules are comprised of multiple cells. We will compare these cases:
+#    - modules with one bypass diode
+#    - modules with three bypass diodes
+#    - half-cut cell modules with three bypass diodes on portrait and landscape
+#
+# Setting up the system
+# ----------------------
+# Let's start by defining the system characteristics, location and the time
+# range for the analysis.
+
+import pvlib
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.dates import ConciseDateFormatter
+
+pitch = 4  # meters
+width = 1.5  # meters
+gcr = width / pitch
+N_modules_per_row = 6
+axis_azimuth = 180  # N-S axis
+axis_tilt = 0  # flat because the axis is perpendicular to the slope
+cross_axis_tilt = -7  # 7 degrees downward to the east
+
+latitude, longitude = 40.2712, -3.7277
+locus = pvlib.location.Location(
+    latitude,
+    longitude,
+    tz="Europe/Madrid",
+    altitude=pvlib.location.lookup_altitude(latitude, longitude),
+)
+
+times = pd.date_range("2001-04-11T03", "2001-04-11T07", periods=24).union(
+    pd.date_range("2001-04-11T16", "2001-04-11T20", periods=24)
+)
+
+# %%
+# True-tracking algorithm and shaded fraction
+# -------------------------------------------
+# Since this model is about row-to-row shading, we will use the true-tracking
+# algorithm to calculate the trackers rotation. Back-tracking reduces the
+# shading between rows, but since this example is about shading, we will not
+# use it.
+#
+# Then, the next step is to calculate the fraction of shaded surface. This is
+# done using :py:func:`pvlib.shading.shaded_fraction1d`. Using this function is
+# straightforward with the variables we already have defined.
+# Then, we can calculate the number of shaded blocks by rounding up the shaded
+# fraction by the number of blocks along the shaded length.
+
+# Calculate solar position to get single-axis tracker rotation and irradiance
+solar_pos = locus.get_solarposition(times)
+solar_apparent_zenith, solar_azimuth = (
+    solar_pos["apparent_zenith"],
+    solar_pos["azimuth"],
+)  # unpack for better readability
+
+tracking_result = pvlib.tracking.singleaxis(
+    apparent_zenith=solar_apparent_zenith,
+    apparent_azimuth=solar_azimuth,
+    axis_tilt=axis_tilt,
+    axis_azimuth=axis_azimuth,
+    max_angle=(-90 + cross_axis_tilt, 90 + cross_axis_tilt),  # (min, max)
+    backtrack=False,
+    gcr=gcr,
+    cross_axis_tilt=cross_axis_tilt,
+)
+
+tracker_theta, aoi, surface_tilt, surface_azimuth = (
+    tracking_result["tracker_theta"],
+    tracking_result["aoi"],
+    tracking_result["surface_tilt"],
+    tracking_result["surface_azimuth"],
+)  # unpack for better readability
+
+# Calculate the tracking angles
+rotation_angle = pvlib.tracking.singleaxis(
```

Comment:
Are both of these `singleaxis()` calls necessary?  Seems like one is redundant with the other.

Reply:
Definitely not, maybe it was Copilot, but it was stupid me for sure.